### PR TITLE
1.0 System introduction

### LINKED ISSUE #202: 1.0 - System

### PR DESCRIPTION
**Pull request by jlndk**
close #202

### CODE REVIEW

File: report/10_System.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+Furthermore none of us in the group has worked on a full-scale ASP.NET app previously and therefore wanted to learn it. 
+
+Furthermore we has containerized our entire application with docker and orchestrated it with docker-compose.
+We almost has almost been able to switch over to a distributed docker swarm cluster, but due to time constraints we didn't finish it so we stuck with our single-machine setup.
+
+The server we have chosen is offered by the cloud provider [DigitalOcean](https://www.digitalocean.com/). We have chosen DigitalOcean since they have a friendly and easy-to-use interface, compared to their competitors, while still providing cheap, stable and powerful products. To be more precise, we run our entire system on a single "droplet" (DigitalOceans name for a virtual server), with 2 CPU cores and 4 GB of RAM. We figured this was a good balance between cost and compute power to serve the amount of traffic we expected to get.
```
Comment (jlndk) [inline report/10_System.md]: Should we move this to the scalability chapter?

Comment (Jakobis) [reply report/10_System.md]: Tror måske den er lidt verbose til en introduktion. Tænker den vigtige del er at vi bruger docker, docker-compose og en enkelt digital ocean server

File: report/10_System.md [hunk 2]
```diff
@@ -1,5 +1,15 @@
 ## System's Perspective
 
-Short Introduction to System goes here.
+<!-- Short Introduction to System goes here.
+Double-check that for all the weekly tasks (those in the end of the lecture notes) you include the corresponding information. -->
 
-Double check that for all the weekly tasks (those in the end of the lecture notes) you include the corresponding information.
+Our web application itself is written in C#, using the ASP.NET web framework.
```
Comment (Jakobis) [inline report/10_System.md]: I think maybe we should try and keep words like "our" out of the report.

File: report/10_System.md [hunk 3]
```diff
@@ -1,5 +1,15 @@
 ## System's Perspective
 
-Short Introduction to System goes here.
+<!-- Short Introduction to System goes here.
+Double-check that for all the weekly tasks (those in the end of the lecture notes) you include the corresponding information. -->
 
-Double check that for all the weekly tasks (those in the end of the lecture notes) you include the corresponding information.
+Our web application itself is written in C#, using the ASP.NET web framework. 
+We have chosen this stack since it is the de-facto-standard for larger web applications (at least in Denmark), which makes it really well supported and maintained. This also means that many third-party packages is available to utilize, like Newtonsoft JSON, which is the de-facto way of (de)serialize JSON, and Prometheus-net for interacting with Prometheus.
```
Comment (Jakobis) [inline report/10_System.md]: Shouldn't the introduction be more about saying what is in the system, and less about why we chose the things we did? It seems like the stuff that should go into separate parts.

**Review by Jakobis — DISMISSED** [review]

**Review by tobloef — CHANGES_REQUESTED** [review]
- "(de)serialize" burde være "to (de)serialize" eller "(de)serializing"
- Er også selv skyldig, men når man skriver uni rapporter burde man ikke skrive "vi", "jeg" og sådan noget. Det er bedre f.eks. at skrive "It was decided" instead of "we decided".

**Comment by Hjaltesorgenfrei** [comment]
>     * Er også selv skyldig, men når man skriver uni rapporter burde man ikke skrive "vi", "jeg" og sådan noget. Det er bedre f.eks. at skrive "It was decided" instead of "we decided".

Where appropriate "our" and "we" is not a problem. But yes it should not be in this introduction to a chapter.

**Comment by Hjaltesorgenfrei** [comment]
Seems overly verbose and takes large parts of what is in the next parts.
So those will probaly have to removed

**Review by Hjaltesorgenfrei — APPROVED** [review]

File: report/10_System.md [hunk 4]
```diff
@@ -1,5 +1,19 @@
...
+Furthermore no one in the group has worked on a full-scale ASP.NET app previously and therefore wanted to learn it. 
 
-Double check that for all the weekly tasks (those in the end of the lecture notes) you include the corresponding information.
+When given the task of rewriting the MiniTwit application a tech stack had to be chosen. 
+This not only includes a programming language and framework, but also which services the application depends on, such as a database, monitoring service, and logging solution.
+To this purpose C#, using the ASP.NET web framework, was chosen as it is well supported and maintained. 
+The framework also has a large community behind it, which also means that many third-party packages are available to utilize, such as Newtonsoft JSON, which is the de-facto way of (de)serialize JSON, and Prometheus-net for interacting with Prometheus.
+Another reason for choosing this tech stack was that no one in the group has worked on a full-scale ASP.NET app previously and therefore wanted to learn it. 
+Since the framework is also backed by Microsoft, which makes the platform viable for businesses, it is a very lucrative skill to have when looking for jobs.
```
Comment (tobloef) [inline report/10_System.md]: I think you forgot to remove this?

**Review by tobloef — APPROVED** [review]

**Review by tobloef — DISMISSED** [review]
oops

**Review by Jakobis — APPROVED** [review]

**Review by Hjaltesorgenfrei — APPROVED** [review]

**Pull request merged**